tests/meshcop/addr_utils.cpp

```diff
@@ -0,0 +1,144 @@
+ *       notice, this list of conditions and the following disclaimer in the
+ *       documentation and/or other materials provided with the distribution.
+ *    3. Neither the name of the copyright holder nor the
+ *       names of its contributors may be used to endorse or promote products
+ *       derived from this software without specific prior written permission.
+ *
+ *    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ *    AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ *    IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ *    ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ *    LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ *    CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ *    SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ *    INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ *    CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *    ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *    POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * @file
+ *   The file is the implementation of the address manipulation utilities for the commissioner test app.
+ */
+
+#include "addr_utils.hpp"
+#include <string.h>
+
+namespace ot {
+namespace BorderRouter {
+
+static const uint8_t kLociid[] = {
+    0x00, 0x00, 0x00, 0xff, 0xfe, 0x00,
+};
+
+enum
+{
+    kRlocRouterIDBitOffset = 10,
+    kRlocAddrUint16Offset  = 7,
+    kRlocAddrUint8Offset   = 14,
+    kAlocRouterByte        = 0xfc,
+    kIidAddrUint8Offset    = 8,
+    kUlaPrefix             = 0xfd,
+};
+
+uint16_t ToRloc16(uint8_t aRouterID, uint16_t aChildID)
+{
+    uint16_t rloc16 = aRouterID;
+
+    return (rloc16 << kRlocRouterIDBitOffset) | aChildID;
+}
+
+#define IPSTR_BUFSIZE (((INET6_ADDRSTRLEN > INET_ADDRSTRLEN) ? INET6_ADDRSTRLEN : INET_ADDRSTRLEN) + 1)
+char *GetIPString(const struct sockaddr *aAddr, char *aOutBuf, size_t aLength)
+{
+    switch (aAddr->sa_family)
+    {
+    case AF_INET:
+        inet_ntop(AF_INET, &(((struct sockaddr_in *)aAddr)->sin_addr), aOutBuf, aLength);
+        break;
+
+    case AF_INET6:
+        inet_ntop(AF_INET6, &(((struct sockaddr_in6 *)aAddr)->sin6_addr), aOutBuf, aLength);
+        break;
+
+    default:
+        strncpy(aOutBuf, "Unknown AF", aLength);
+        return NULL;
+    }
+
+    return aOutBuf;
+}
+
+struct in6_addr ConcatRloc16Address(const in6_addr &aPrefix, uint16_t aRloc16)
+{
+    in6_addr addr = aPrefix;
+
+    reinterpret_cast<uint16_t *>(&addr)[kRlocAddrUint16Offset] = htons(aRloc16);
+    return addr;
+}
+
+struct in6_addr ConcatRloc16Address(const in6_addr &aPrefix, uint8_t aRouterID, uint16_t aChildID)
+{
+    return ConcatRloc16Address(aPrefix, ToRloc16(aRouterID, aChildID));
+}
+
+struct in6_addr FindRloc16Address(const std::vector<struct in6_addr> &aAddrs)
```

Comment:
Are `FindRloc16Address` and `FindMLEIDAddress`, `GetRlocPrefix` actually used in this PR?